coffeecard/CoffeeCard.WebApi/Models/Statistic.cs

```diff
@@ -25,16 +25,13 @@ public class Statistic
 
         public static DateTime GetSemesterStart(DateTime currentTime)
         {
-            if (currentTime.Month < 7)
-            {
-                var jan1 = new DateTime(currentTime.Year, 1, 1);
-                var dayOffset = DayOfWeek.Monday - jan1.DayOfWeek;
-                var startDate = 29 - dayOffset;
-
-                return new DateTime(currentTime.Year, 1, startDate);
-            }
+            // Autumn semester: Get first day of July.
+            if (currentTime.Month >= 7) return new DateTime(currentTime.Year, 7, 1);
 
-            return new DateTime(currentTime.Year, 7, 1);
+            // Spring semester: Get last Monday of January.
+            var lastDayOfJan = new DateTime(currentTime.Year, 1, 31);
+            int correctedLastDayOfWeek = (int)(lastDayOfJan.DayOfWeek + 6) % 7; // Mon=0, Tue=1, ..., Sun=6
+            return lastDayOfJan.AddDays(-correctedLastDayOfWeek);
         }
 
         public static DateTime GetSemesterEnd(DateTime currentTime)
```

Comment:
Let's add some documentation while we're here. We could also unit test this one but will not require this of you 😉 

```suggestion
        /// <summary>
        /// Get the end of date of the semester.
        /// When <paramref name="currentTime"/> is in July (Month 7) or greater, the semester start is December 23rd or when earlier than July, the semester start is June 30rd.
        /// </summary>
        public static DateTime GetSemesterEnd(DateTime currentTime)
```